2008-06-27  Avery Pennarun <averyp@averyp-server>

	* Initialize an otherwise uninitialized variable in bn_mont.c.  This
	  problem was causing lots and lots of valgrind errors in wvstreams.

diff -rpuN --exclude .depend --exclude '*~' --exclude '*.orig' --exclude '*.rej' --exclude '*.l[oa]' --exclude '*.lai' --exclude '*.o' --exclude .deps --exclude .libs ../.build_orig/openssl/crypto/bn/bn_lib.c openssl/crypto/bn/bn_lib.c
--- ../.build_orig/openssl/crypto/bn/bn_lib.c	2007-03-27 20:14:18.000000000 -0400
+++ openssl/crypto/bn/bn_lib.c	2008-06-27 20:11:54.000000000 -0400
@@ -304,6 +304,7 @@ BIGNUM *BN_new(void)
 		BNerr(BN_F_BN_NEW,ERR_R_MALLOC_FAILURE);
 		return(NULL);
 		}
+	memset(ret, 0, sizeof(BIGNUM));
 	ret->flags=BN_FLG_MALLOCED;
 	ret->top=0;
 	ret->neg=0;
